docs/articles/clustering/cluster-sharding.md

```diff
@@ -257,3 +257,188 @@ In the normal operation of an Akka.NET cluster, the sharding system automaticall
 However, in the event that an `ActorSystem` is aborted as a result of a process / hardware failure it's possible that when using `akka.cluster.sharding.state-store-mode=persistence` leftover sharding data can still be present inside the Akka.Persistence journal and snapshot store - which will prevent the Akka.Cluster.Sharding system from recovering and starting up correctly the next time it's launched.
 
 This is a _rare_, but not impossible occurrence. In the event that this happens you'll need to purge the old Akka.Cluster.Sharding data before restarting the sharding system. You can purge this data automatically by [using the Akka.Cluster.Sharding.RepairTool](https://github.com/petabridge/Akka.Cluster.Sharding.RepairTool) produced by [Petabridge](https://petabridge.com/).
+
+## Tutorial
+
+In this tutorial, we will be making a very simple non-persisted shopping cart implementation using cluster sharding.
+All the code used in this tutorial can be found in the [GitHub repository](https://github.com/akkadotnet/akka.net/tree/dev/src/examples/Cluster/ClusterSharding/ShoppingCart)
+
+For a sample of a distributed data backed persistent example, please see [this example project](https://github.com/akkadotnet/akka.net/tree/dev/src/examples/Cluster/ClusterSharding/ClusterSharding.Node)
+instead.
+
+### Setting Up the Roles
+
+In a sharded cluster, it is common for the shards to be assigned their own specialized role inside the
+cluster to distribute their actors in. In this tutorial we will have a single frontend node that will
+feed three backend nodes with purchasing data. Usually, these nodes will be separated into different
+specialized projects but in this example, we will roll them into a single project and control their
+roles using an environment variable.
+
+```c#
+using Akka.Actor;
+using Akka.Bootstrap.Docker;
+using Akka.Configuration;
+
+var role = Environment.GetEnvironmentVariable("IS_FRONTEND") == "true" ? FrontEndRole : BackEndRole;
+
+var config = ConfigurationFactory.ParseString(@$"
+        akka.actor.provider = cluster
+        akka.cluster.roles=[{role}]
+        akka.cluster.min-nr-of-members = 4")
+    .BootstrapFromDocker();
+
+var system = ActorSystem.Create("shopping-cart", config);    
+```
+
+### Starting Up Cluster Sharding
+
+Cluster sharding can be added by using the `Akka.Cluster.Sharding` NuGet package, it already contains
+references to the other required packages. Note that the `ClusterSharding.Get()` call is very important
+as it contains all the initialization code needed by cluster sharding to start. Note that all nodes that
+participates or interacts with the sharded cluster will need to initialize ClusterSharding.
+
+```c#
+using Akka.Cluster.Sharding;
+
+var sharding = ClusterSharding.Get(system);
+```
+
+### Starting the Sharding Coordinator Actors
+
+There are two types of sharding coordinator actors; a normal coordinator coordinates messages and
```

Comment:
Use a bulleted list to spell out both types of actors and make it clear that if you have a `ShardRegion` you don't also need a `ShardRegionProxy` to communicate with those sharded actors